common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/DataFlowTaskLaunchRequestAutoConfiguration.java

```diff
@@ -144,15 +157,14 @@ public TaskLaunchRequestMessageProcessor taskLaunchRequestMessageProcessor(
 
 		@Override
 		public Collection<String> processMessage(Message<?> message) {
-			return evaluateArgExpressions(message, argExpressionsMap);
+			return evaluateArgExpressions(message);
 		}
 
-		private Collection<String> evaluateArgExpressions(Message<?> message, Map<String, Expression> argExpressions) {
+		private Collection<String> evaluateArgExpressions(Message<?> message) {
 			List<String> results = new LinkedList<>();
-			argExpressions.forEach((k, v) -> {
-				Expression expression = argExpressionsMap.get(k);
-				Assert.notNull(expression, String.format("expression %s cannot be null!", v));
-				results.add(String.format("%s=%s", k, expression.getValue(message)));
+			this.argExpressionsMap.forEach((k, expression) -> {
+				Assert.notNull(expression, String.format("expression %s cannot be null!", expression));
```

Comment:
??? `expression null cannot be null!` ?
What is the point placing `null` word into the sentence if you are going to say that it can't be null? 😄

Reply:
good catch